### PR TITLE
Jenkins AWS smoke test

### PR DESCRIPTION
**Pull request by ethernetdan**
Adds AWS smoke testing to be eventually run on PRs

Creates a new folder for test configuration but we may want to get it working with the examples folder.

### CODE REVIEW

**Comment by Quentin-M** [comment]
Have you seen https://github.com/coreos/tectonic-installer/issues/61?

File: Jenkinsfile [hunk 1]
```diff
@@ -17,5 +17,54 @@ pipeline {
         sh 'make structure-check'
       }
     }
+
+    stage('Smoke Tests') {
+      steps {
+        parallel (
+          "AWS": {
+              withCredentials([file(credentialsId: 'tectonic-license', variable: 'TF_VAR_tectonic_pull_secret_path'),
```
Comment (dghubble) [inline Jenkinsfile]: @ethernetdan If you check the Tectonic repo's Jenkinsfile, there is a tip about avoiding writing logic like this in the Jenkinsfile directly. Can you place this in a script which gets invoked please?

File: Jenkinsfile [hunk 2]
```diff
@@ -17,5 +17,54 @@ pipeline {
...
+
+              # Create local config
+              make localconfig
+
+              # Use smoke test configuration for deployment
+              ln -sf ${WORKSPACE}/test/aws.tfvars ${WORKSPACE}/build/${CLUSTER}/terraform.tfvars
+
+              make plan
+
+              # always cleanup cluster
```
Comment (dghubble) [inline Jenkinsfile]: You can use the Jenkins declarative pipeline `post` feature as well, to ensure a certain cleanup script or task is always run at the Jenkins level.

File: Makefile [hunk 3]
```diff
@@ -26,7 +26,7 @@ apply: $(BUILD_DIR)/.terraform
 	cd $(BUILD_DIR) && terraform apply $(TOP_DIR)/platforms/$(PLATFORM)
 
 destroy:
-	cd $(BUILD_DIR) && terraform destroy $(TOP_DIR)/platforms/$(PLATFORM)
+	cd $(BUILD_DIR) && terraform destroy -force $(TOP_DIR)/platforms/$(PLATFORM)
```
Comment (mxinden) [inline Makefile]: @ethernetdan thanks! Will need this for the prometheus operator e2e tests. :)

File: Jenkinsfile (hunk 1 above)
Comment (mxinden) [inline Jenkinsfile]: Having this in two scripts (e.g. 'bring-up-cluster', 'tear-down-cluster') would help us a lot with testing the prometheus operator.

**Comment by mxinden** [comment]
@ethernetdan Thanks for this! We are about to use the tectonic-installer to bring up a cluster in the prometheus-operator e2e-tests. It works so far ([tectonic-installer-branch](https://github.com/mxinden/prometheus-operator/tree/tectonic-installer) & [jenkins-job](https://jenkins-monitoring.prod.coreos.systems/job/prometheus-operator-e2e-tests-tectonic-installer/lastFailedBuild/console)), but  is still  hacky. I will adapt most of this as it is a lot cleaner.

**Comment by alexsomesan** [comment]
@dghubble I remember your suggestion from earlier on about following the example of the current tectonic Jenkinsfile to use K8S pods as workers. That's actually the path I had set out on, but have since heard from @chancez and @bison that we can't actually use that approach for open-source projects, because of security concerns around PR form outside CoreOS.
Given those restrictions, I think we need to keep the current setup until we can maybe configure an isolated cluster for these workers.

File: Jenkinsfile (hunk 2 above)
Comment (alexsomesan) [inline Jenkinsfile]: That's a good point! We should make use of the post step for cleanup.

**Comment by dghubble** [comment]
@alexsomesan external contributors should not be running un-approved code on that CoreOS VM via docker or in pods on CoreOS cluster infrastructure. Neither is safe.

**Pull request merged**